tsl/src/continuous_aggs/create.c

```diff
@@ -1852,6 +1850,10 @@ cagg_update_view_definition(ContinuousAgg *agg, Hypertable *mat_ht,
 		view_tle->resname = user_tle->resname;
 	}
 
+	/* keep locks until end of transaction */
+	relation_close(direct_view_rel, NoLock);
```

Comment:
I believe this one doesn't need to be moved, since the object is copied. See https://github.com/timescale/timescaledb/pull/2043/files#diff-fee5e1abf06aea290bbb4493a38113a9R1815

Reply:
well the previous position was kinda arbitrary and i like having them together at the bottom, but it could also be moved to the line after the copy

Reply:
I agree this one was at arbitrary place. At least I follow the code :) Since the lock is still kept, it doesn't matter how early the relation is closed.